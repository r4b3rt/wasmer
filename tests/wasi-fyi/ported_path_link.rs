// WASI:
// mapdir: act5:test_fs/hamlet/act5
// tempdir: temp

use std::fs;
use std::io::Read;

fn main() {
    {
        std::fs::hard_link("/hamlet/act5/scene1.txt", "/tmp/scene_of_the_day.txt").unwrap();
        let mut f = fs::OpenOptions::new()
            .read(true)
            .open("/tmp/scene_of_the_day.txt")
            .unwrap();
        let mut buffer = [0u8; 64];
        f.read_exact(&mut buffer).unwrap();

        println!("{}", std::str::from_utf8(&buffer[..]).unwrap());
        for b in buffer.iter_mut() {
            *b = 0;
        }

        let mut f = fs::OpenOptions::new()
            .read(true)
            .open("/hamlet/act5/scene1.txt")
            .unwrap();
        f.read_exact(&mut buffer).unwrap();
        println!("{}", std::str::from_utf8(&buffer[..]).unwrap());
    }

    std::fs::remove_file("/tmp/scene_of_the_day.txt").unwrap();
    let path = std::path::PathBuf::from("act5/scene1.txt");

    if path.exists() {
        println!("Path still exists");
    } else {
        println!("Path was deleted!");
    }
}
