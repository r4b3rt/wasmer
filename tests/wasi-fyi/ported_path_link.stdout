ACT V
SCENE I. A churchyard.

    Enter two Clowns, with spades,
ACT V
SCENE I. A churchyard.

    Enter two Clowns, with spades,
Path was deleted!
